server/sonar-server/src/main/java/org/sonar/server/ui/ws/SettingsNavigationAction.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.ui.ws;
+
+import org.sonar.api.config.Settings;
+import org.sonar.api.i18n.I18n;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService.NewController;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.api.web.NavigationSection;
+import org.sonar.api.web.Page;
+import org.sonar.core.permission.GlobalPermissions;
+import org.sonar.server.plugins.UpdateCenterClient;
+import org.sonar.server.ui.ViewProxy;
+import org.sonar.server.ui.Views;
+import org.sonar.server.user.UserSession;
+
+public class SettingsNavigationAction implements NavigationAction {
+
+  private final Settings settings;
+  private final Views views;
+  private final I18n i18n;
+
+  public SettingsNavigationAction(Settings settings, Views views, I18n i18n) {
+    this.views = views;
+    this.settings = settings;
+    this.i18n = i18n;
+  }
+
+  @Override
+  public void define(NewController context) {
+    context.createAction("settings")
+      .setDescription("Get information concerning settings navigation for the current user.")
```

Comment:
The description is misleading : it makes me think that it's returning all settings pages (pages under Configuration, Security, System) but it's only returning settings pages provided by plugin and it's also saying if the update center page should be displayed or not.

I would have implemented this WS to return all settings pages, but if you keep current implementation please update the description.